core/embed/rust/src/ui/model_tt/layout.rs

```diff
@@ -666,6 +680,52 @@ extern "C" fn new_show_error(n_args: usize, args: *const Obj, kwargs: *mut Map)
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
+extern "C" fn new_confirm_webauthn(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+    let block = move |_args: &[Obj], kwargs: &Map| {
+        let title: StrBuffer = kwargs.get(Qstr::MP_QSTR_title)?.try_into()?;
+        let icon: Option<StrBuffer> = kwargs.get(Qstr::MP_QSTR_icon)?.try_into_option()?;
+        let pages: Obj = kwargs.get(Qstr::MP_QSTR_pages)?;
+
+        // Need to find the overall page count for pagination purposes.
+        let page_count = {
+            let mut iter_buf = IterBuf::new();
+            let iter = Iter::try_from_obj_with_buf(pages, &mut iter_buf)?;
+            iter.count()
+        };
+
+        // Closure to lazy-load the information on given page index.
+        // Done like this to allow arbitrarily many pages without
+        // the need of any allocation here in Rust.
+        let get_page = move |page_index| {
+            let page: Obj = {
+                let mut iter_buf = IterBuf::new();
+                let mut iter = unwrap!(Iter::try_from_obj_with_buf(pages, &mut iter_buf));
+                iter.nth(page_index).unwrap()
+            };
+
+            let [app_name, account_name]: [StrBuffer; 2] = unwrap!(iter_into_array(page));
+
+            FidoPage::new(app_name, account_name)
+        };
+
+        let controls = Button::cancel_confirm(
+            Button::with_icon(theme::ICON_CANCEL).styled(theme::button_cancel()),
+            Button::with_text("CONFIRM").styled(theme::button_confirm()),
+            2,
+        );
+
+        let fido_page = FidoPaginatedPage::new(get_page, page_count, icon, controls);
+
+        let obj = LayoutObj::new(
+            Frame::new(title, fido_page)
+                .with_border(theme::borders())
+                .into_child(),
```

Comment:
Turns out I lied and LayoutObj::new wraps the component in Child<> automatically, sorry. See also #2581 #2600.